### PR TITLE
unify and tidy objstore setup

### PR DESCRIPTION
**Pull request by Nicholaswang**
1, add NewBucketWithConfig func for azure, bos and oss to keep updated with cos and s3
2, add config validate func for oss

### CODE REVIEW

**Comment by Nicholaswang** [comment]
@hanjm I submitted a new pr, there was an extra space in the oss.go file, the go-lint result was somewhat misleading which indicated that was related with an unstaged file. Now the checks have all been passed. Please review the pr again.

**Comment by hanjm** [comment]
👍  LGTM

File: pkg/objstore/gcs/gcs.go [hunk 1]
```diff
@@ -52,6 +52,7 @@ func NewBucket(ctx context.Context, logger log.Logger, conf []byte, component st
 	return NewBucketWithConfig(ctx, logger, gc, component)
 }
 
+// NewBucketWithCoonfig returns a new Bucket with gcs Config struct.
```
Comment (GiedriusS) [inline pkg/objstore/gcs/gcs.go:55]: Nit:
```suggestion
// NewBucketWithConfig returns a new Bucket with gcs Config struct.
```

**Review by GiedriusS — DISMISSED** [review]
LGTM except for one nit

**Review by GiedriusS — APPROVED** [review]

**Pull request merged**